### DISCUSSION TITLE
next with react-native-web Error import typeof AccessibilityInfo from './Libraries/Components/AccessibilityInfo/AccessibilityInfo';

### DISCUSSION THREAD

**Discussion by francisleigh** [vercel/next.js#18029] [Help]
# Bug report

## Describe the bug
I am using my own react-native npm [module](https://www.npmjs.com/package/maisonsport-common-ui) that is consumed by my `next.js with-react-native-for-web` App. When running `npm run dev` i get an error: 
```bash
import typeof AccessibilityInfo from './Libraries/Components/AccessibilityInfo/AccessibilityInfo';
^^^^^^

SyntaxError: Cannot use import statement outside a module
```
## To Reproduce
1. create a with-react-native-web next project
2. `npm i maisonsport-common-ui`
3. `import { Text } from 'maisonsport-common-ui';` into `pages/index.js`
4. `npm run dev`
4. See error (if error doesn't show at first then reload the localhost page)

## Expected behavior
I should be able to use npm package seamlessly in the browser.

## Screenshots
![Screen-Recording-2020-10-19-at-2](https://user-images.githubusercontent.com/10357801/96508214-28705200-1252-11eb-9f7d-ceeda39c0431.gif)
^ Here illustrated a weird behavior where I can get the code to run when hot-reloading but when I do a full browser refresh the error appears again.

## System information

- OS: macOS
- Version of Next.js: `"next": "latest",` (installed today)
- Version of Node.js: v14.8.0

## next.config.js
```javascript
const withTM = require('next-transpile-modules')(['maisonsport-common-ui']);

module.exports = withTM({
  webpack: (config) => {
    config.resolve.alias = {
      ...(config.resolve.alias || {}),
      // Transform all direct `react-native` imports to `react-native-web`
      'react-native$': 'react-native-web',
    }
    config.resolve.extensions = [
      '.web.ts',
      '.web.tsx',
      '.ts',
      '.tsx',
      '.web.js',
      '.web.jsx',
      '.js',
      '.jsx',
      ...config.resolve.extensions,
    ]

    return config
  },
})

```

**Comment by ljosberinn** [comment]
`import typeof` looks like flow syntax for which you probably need [something like this example does](https://github.com/vercel/next.js/blob/canary/examples/with-flow/package.json)

**Comment by mwarger** [comment]
I'm seeing this in a project as well when using emotion/native.  Common between these seems to be that they require('react-native').  Maybe at some point when reloading the project, the files stop getting compiled?  I'm not sure about this.

**Comment by etoal83** [comment]
I bumped into the same error in my app using NextJS and react-native-web as well, then I could finally solve it. In my case, the @mwarger 's suggestion helped me to notice that the error comes from an untranspiled module (actually that is @emotion/native in my case). I simply added the module name into the transpilation list like this:

```javascript
const withTM = require('next-transpile-modules')([
  '@emotion/native',
]);
```

Then the error went away. (@mwarger thanks so much!)

Though I'm a beginner in coding so having less understanding of transpilation or ES modules, I also wonder why your first line in `next.config.js` could not solve the error...

**Reply by dante022** [reply]
May I ask you how did you find which one is untranspiled in your case?

I'm in same error with this but I can't find which modules are untranspiled

**Reply by warent** [reply]
For future reference anyone else wondering about this, the error told me the directory within node_modules with the file that was having the problem. i.e. the directory name is the package that needs to be added to the list
